src/Microsoft.OData.Edm/Csdl/Parsing/AnnotationJsonParser.cs

```diff
@@ -0,0 +1,515 @@
+            {
+                return false;
+            }
+
+            CsdlExpressionBase expression = propertyValue.ProcessProperty("$LabeledElement", context, ParseExpression);
+
+            // a member $Name whose value is a string containing the labeled element�s name.
+            string label = element.ProcessRequiredProperty("$Name", context, (v, p) => v.ParseAsString(p));
+
+            labeledExp = new CsdlLabeledExpression(label, expression, context.Location());
+            return true;
+        }
+
+        /// <summary>
+        /// Parse the Labeled Element Reference expression using <see cref="JsonElement"/>.
+        /// </summary>
+        /// <param name="element">The input JSON element.</param>
+        /// <param name="context">The parser context.</param>
+        /// <param name="labeledReferenceExp">The built Labeled Element Reference expression.</param>
+        /// <returns>true/false</returns>
+        private static bool TryParseLabeledElementReferenceExpression(JsonElement element, JsonParserContext context, out CsdlLabeledExpressionReferenceExpression labeledReferenceExp)
+        {
+            Debug.Assert(context != null);
+            labeledReferenceExp = null;
+
+            // Labeled element reference expressions are represented as an object with a member $LabeledElementReference whose value is a string containing an qualified name.
+            JsonElement propertyValue;
+            if (element.ValueKind != JsonValueKind.Object || !element.TryGetProperty("$LabeledElementReference", out propertyValue))
+            {
+                return false;
+            }
+
+            string label = propertyValue.ProcessProperty("$LabeledElementReference", context, (e, c) => e.ParseAsString(c));
+
+            labeledReferenceExp = new CsdlLabeledExpressionReferenceExpression(label, context.Location());
+            return true;
+        }
+
+        /// <summary>
+        /// Parse the record expression expression using <see cref="JsonElement"/>.
+        /// </summary>
+        /// <param name="element">The input JSON element.</param>
+        /// <param name="context">The parser context.</param>
+        /// <returns>the built record expression.</returns>
+        private static CsdlRecordExpression ParseRecordExpression(JsonElement element, JsonParserContext context)
+        {
+            Debug.Assert(context != null);
+
+            // A record expression MAY specify the structured type of its result, which MUST be an entity type or complex type in scope.
+            // If not explicitly specified, the type is derived from the expression�s context
+
+            // The type of a record expression is represented as the @type control information
+            // for example: "@type": "https://example.org/vocabs/person#org.example.person.Manager",
+            // So far, ODL doesn't support the type "@type" with relative path, only supports like "#Model.VipCustomer", or without #

```

Comment:
>So far, ODL doesn't support the type "@type" with relative path [](start = 15, length = 63)

Is this documented somewhere? Do we have a work item to track?

Reply:
Filed an issue at [https://github.com/OData/odata.net/issues/1821](https://github.com/OData/odata.net/issues/1821) to track on it.

---
In reply to: [443017753](https://github.com/OData/odata.net/pull/1775#discussion_r443017753) [](ancestors = 443017753)

Reply:
thanks.

---
In reply to: [444545165](https://github.com/OData/odata.net/pull/1775#discussion_r444545165) [](ancestors = 444545165,443017753)